dali/operators/generic/slice/subscript.h

```diff
@@ -0,0 +1,325 @@
+        int64_t lo = s.lo.IsDefined() ? s.lo.values[i] : 0;
+        int64_t hi = s.hi.IsDefined() ? s.hi.values[i] : in_extent;
+        int64_t step = s.step.IsDefined() ? s.step.values[i] : 1;
+        DALI_ENFORCE(step == 1, "Indexing with non-unit step is not implemented");
+        if (lo < 0) lo += in_extent;
+        if (hi < 0) hi += in_extent;
+        lo = clamp(lo, 0_i64, in_extent);
+        hi = clamp(hi, 0_i64, in_extent);
+        lo_.tensor_shape_span(i)[d] = lo;
+        hi_.tensor_shape_span(i)[d] = hi;
+        step_.tensor_shape_span(i)[d] = step;
+
+        int64_t out_extent = step > 0 ? div_ceil(hi - lo,  step)
+                                      : div_ceil(lo - hi, -step);
+        if (out_extent < 0)
+          out_extent = 0;
+        shape_.tensor_shape_span(i)[d] = out_extent;
+      }
+    }
+  }
+
+  void ProcessShapes(TensorListShape<> &out_shape, const TensorListShape<> &in_shape) {
+    int in_dims = in_shape.sample_dim();
+    int nsub = subscripts_.size();
+
+    out_dim_map_.clear();
+    dim_groups_.clear();
+
+    int group_start = 0;
+    int d = 0;
+    for (; d < nsub; d++) {
+      if (subscripts_[d].IsDefined()) {
+        if (d != group_start)
+          dim_groups_.push_back({ group_start, d - group_start });
+        group_start = d + 1;
+      }
+
+      if (!subscripts_[d].IsIndex()) {  // indices are not present in the output
+        out_dim_map_.push_back(d);
+      }
+    }
+    for (; d < in_dims; d++) {
+      out_dim_map_.push_back(d);
+    }
+
+    if (in_dims != group_start)
+      dim_groups_.push_back({ group_start, in_dims - group_start });
+
+    collapse_dims(simplified_in_shape_, in_shape, dim_groups_);
+    collapse_dims(simplified_out_shape_, shape_, dim_groups_);
+    collapse_dims(simplified_anchor_, lo_, dim_groups_);
+
+    out_shape.resize(in_shape.num_samples(), out_dim_map_.size());
+    for (int i = 0; i < out_shape.num_samples(); i++) {
+      auto out_sample = out_shape.tensor_shape_span(i);
+      auto sh = shape_.tensor_shape_span(i);
+      for (int d = 0; d < out_shape.sample_dim(); d++) {
+        out_sample[d] = sh[out_dim_map_[d]];
+      }
+    }
+  }
+
+  TensorLayout GetOutputLayout(const TensorLayout &input_layout) const {
+    if (input_layout.empty())
+      return {};
+    TensorLayout out_layout;
+    out_layout.resize(out_dim_map_.size());
+    for (int i = 0; i < out_layout.ndim(); i++)
+      out_layout[i] = input_layout[out_dim_map_[i]];
+    return out_layout;
+  }
+
+  bool CanInferOutputs() const override { return true; }
+
+  using Operator<Backend>::RunImpl;
+  void RunImpl(workspace_t<Backend> &ws) override {
+    const auto &input = ws.template InputRef<Backend>(0);
+    auto &output = ws.template OutputRef<Backend>(0);
+    output.SetLayout(GetOutputLayout(input.GetLayout()));
+    VALUE_SWITCH(simplified_in_shape_.sample_dim(), ndim,
+      (1,  2,  3,  4,  5/*,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16*/),
```

Comment:
You can add a comment why it is commented out.

Reply:
Mhh... It shouldn't be.  It's commented to save time in local builds (and it's _very_ slow to build).  I hope I remember to uncomment it in the final commit.